rust/src/dataset.rs

```diff
@@ -684,8 +684,7 @@ impl Dataset {
     }
 
     /// Delete rows based on a predicate.
-    #[allow(dead_code)] // TODO: remove once this is public
-    pub(crate) async fn delete(&mut self, predicate: &str) -> Result<()> {
+    pub async fn delete(&mut self, predicate: &str) -> Result<()> {
```

Comment:
Because of this... we shouldn't merge this PR. But we can only use public functions in benchmarks.

@chebbyChefNEQ do you want to merge this branch into #958 and we can iterate there?

Reply:
#958 has been merged